sstables/sstable_set.cc

```diff
@@ -464,6 +470,104 @@ class time_series_sstable_set : public sstable_set_impl {
     }
 };
 
+// Reader and owning pointer to the sstable being read.
+// Prevents the sstable from being closed while the reader is being used.
```

Comment:
Unnecessary, the sstable reader already holds on the a shared pointer.

Reply:
You mean whoever calls `create_single_key_sstable_reader` is keeping the sstable set alive for the whole duration of the query? It's not obvious from the code and nowhere documented. But yeah, if that's true, I can simplify this.

Reply:
The `sstable_reader`, thats doing the actual reading is keeping the sstable alive.

Reply:
Ahh, ok, got it.